[feature][backward incompatible] introduce monit_config, state in monit_rc, monit_conf_extra_include_dir

trombik opened the pull request:
this PR introduces flexible `monit_config` and is backward incompatible.

it also introduces,

* `state` in monit_rc, which enables to remove fragments
* `monit_conf_extra_include_dir`, which optionally allows users to include other directories

Commit:
[feature][backward incompatible] introduce monit_config

Commit:
[feature][backward incompatible] introduce `state`

Commit:
[feature] introduce monit_conf_extra_include_dir

Commit:
[documentation] clarify include directives

Commit:
update README

Commit:
[bugfix] fix the centos build

Commit:
[documentation] document incompatibilities

Commit:
ditch deprecated variables

mitsururike commented on README.md:
```diff
@@ -26,7 +27,45 @@ None
+| `monit_rc` | dict of `monit` configuration fragment files (see below) | `{}` |
+| `monit_config` | string of `monit_conf_file` (see below) | `""` |
+
+## `monit_conf_include_dir`
```
* typo: s/monit_conf_include_dir/monit_conf_extra_include_dir/ ?
* this variable is an array and the name should be plural form to clarify that it is not a single value.
* i do not understand the necessity of the variable from the example.  ~~if the directory is not created by the task 
 "Create monit_conf_extra_include_dir directories" but by the other project, i would understand the necessity.~~ 
* the directory should be created by the other project. the other project which create a file to include should be responsible for creating the directory as well.

trombik replied:
would you elaborate the last question? i cannot decipher it.

mitsururike replied:
my understanding is that the other project creates files to be included by monit in the directories in `monit_conf_extra_include_dir`. the directories should be created by the project which creates
the files, not by the task "Create monit_conf_extra_include_dir directories"

mitsururike commented on tasks/main.yml:
```diff
@@ -3,24 +3,88 @@
+    that:
+      - "'path' in item"
+      - item.path is string
+
```
the blank line should be AFTER the "with_items" statement.

Commit:
[bugfix] wrong new line

Commit:
[bugfix] typo

mitsururike commented on tests/serverspec/default_spec.rb:
```diff
@@ -9,31 +9,50 @@
 ssh_rc_command = "/etc/init.d/ssh"
+default_user = "root"
+default_group = "root"
+extra_user = ""
```
`extra_user = "www"` and `extra_group = "www"` could remove the lines in "freebsd" and in "openbsd".

trombik replied:
is this a showstopper?

mitsururike replied:
the default value should be the largest common.

trombik replied:
although no answer, am assuming the answer is positive.

would you create a rule for it?

Commit:
rename

s/monit_conf_extra_include_dir/monit_conf_extra_include_directories/g

requested at
https://github.com/reallyenglish/ansible-role-monit/pull/16#discussion_r136001735

Commit:
do not create monit_conf_extra_include_directories

Commit:
replace the defaults

requested at:
https://github.com/reallyenglish/ansible-role-monit/pull/16#discussion_r136010485

Commit:
QA

Commit:
recover unexpectedly removed line

fixes the Jenkins build

Commit:
typo

mitsururike commented on CHANGELOG.md:
```diff
@@ -25,7 +25,7 @@ The following variables has gone in this release.
 * `monit_conf_httpd_allow`
 
-If they should be removed and ported using `monit_config`.
+They should be removed and ported using `monit_config`.
```
s/should/must/ ?

Commit:
s/should/must/

mitsururike commented on README.md:
```diff
@@ -79,17 +115,40 @@ None
+    monit_conf_extra_include_directories:
+      - path: /usr/local/project/config/monit
+        state: enabled
+        mode: 750
```
"mode", "owner" and "group" are not required as the directory is managed by the other.

mitsururike replied:
this has not been fixed.
in addition, "extra_user" and "extra_group" are not used.

mitsururike commented on README.md:
```diff
@@ -26,7 +27,42 @@ None
+## `monit_conf_extra_include_directories`
+
+This is a list of dict. The dict is described below. Each `path` to a directory
+is created with the specified `owner`, `group`, and `mode`, and is included by
```
the directory is NOT created.

Commit:
remove obsolete variables from the playbook

and update README

mitsururike commented on tests/serverspec/redhat.yml:
```diff
@@ -3,17 +3,35 @@
+    monit_conf_extra_include_directories:
+      - path: /usr/local/project/config/monit
+        state: enabled
+        mode: 750
```
"mode", "owner" and "group" are not used.

Commit:
remove obsolete variables from the playbook from redhat

Commit:
remove unused variables from playbooks

and update README

mitsururike commented on tests/serverspec/default_spec.rb:
```diff
@@ -48,10 +56,29 @@
+  it { should be_mode 755 }
+end
+
+describe file("/no/such/dir") do
```
the test is not necessary because the role has been modified not to create directories.

Commit:
remove useless test

mitsururike commented on defaults/main.yml:
```diff
@@ -4,20 +4,11 @@ monit_service: monit
 monit_conf_dir: "{{ __monit_conf_dir }}"
 monit_conf_file: "{{ monit_conf_dir }}/monitrc"
 monit_conf_include_dir: "{{ __monit_conf_include_dir }}"
+monit_conf_extra_include_directoriesectories: []
```
typo: s/directoriesectories/directories/

Commit:
typo
